sync/awsdatabricks.py

```diff
@@ -4,30 +4,62 @@
 import io
 import logging
 import zipfile
+from pathlib import Path
 from time import sleep
 from typing import Any, TypeVar
 from urllib.parse import urlparse
 
 import boto3 as boto
 
+import sync.clients.databricks as dbx
 from sync.api.predictions import create_prediction_with_eventlog_bytes, get_prediction
 from sync.api.projects import get_project
-from sync.clients.databricks import get_default_client
 from sync.config import CONFIG
 from sync.models import DatabricksAPIError, DatabricksError, Platform, Response
 
 logger = logging.getLogger(__name__)
 
 
 def create_prediction(
-    plan_type: str, compute_type: str, eventlog: bytes, project_id: str = None
+    plan_type: str,
+    compute_type: str,
+    cluster: dict,
+    cluster_events: dict,
+    instances: dict,
+    eventlog: bytes,
+    project_id: str = None,
 ) -> Response[str]:
     """Create a Databricks prediction
 
     :param plan_type: either "Standard", "Premium" or "Enterprise"
     :type plan_type: str
     :param compute_type: e.g. "Jobs Compute"
     :type compute_type: str
+
+    :param cluster: The Databricks cluster definition as defined by -
+        https://docs.databricks.com/dev-tools/api/latest/clusters.html#get
+    :type cluster: dict
+
+    :param cluster_events: All events, including paginated events, for the cluster as defined by -
+        https://docs.databricks.com/dev-tools/api/latest/clusters.html#events
+
+        If the cluster is a long-running cluster, this should only include events relevant to the time window that a
+        run occurred in.
+    :type cluster_events: dict
+
+    :param instances: All EC2 Instances that were a part of the cluster. Expects a data format as is returned by
+        boto3's EC2.describe_instances API - https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2/client/describe_instances.html
+
+        Instances should be narrowed to just those instances relevant to the Databricks Run. This can be done by passing
+        a `tag:ClusterId` filter to the describe_instances call like -
+        Filters=[
+            {"Name": "tag:ClusterId", "Values": ["my-dbx-clusterid"]}
+        ]
+
+        If there are multiple pages of instances, all pages should be accumulated into 1 dictionary and passed to this
+        function
+    :type instances: dict
+
```

Comment:
Getting an unexpected indentation warning when trying to build the docs that seems to come from this. Also, I'm not sure I'm a fan of the blank lines between parameters in the interest of screen space, inconsistency with autogenerated docstrings and presentation is more of a concern for sphinx.